cypress/integration/work-order/update-with-adding-multiple-sors.spec.js

```diff
@@ -0,0 +1,54 @@
+describe('Updating a work order', () => {
+  context('Adding multiple SORs', () => {
+    beforeEach(() => {
+      cy.loginWithContractorRole()
+
+      cy.fixture('workOrders/workOrder.json')
+        .then((workOrder) => {
+          workOrder.reference = 10000040
+          cy.intercept(
+            { method: 'GET', path: '/api/workOrders/10000040' },
+            { body: workOrder }
+          )
+        })
+        .as('workOrder')
+
+      cy.fixture('workOrders/tasksAndSors.json')
+        .then((tasksAndSors) => {
+          tasksAndSors.splice(1, 2)
+
+          cy.intercept(
+            { method: 'GET', path: '/api/workOrders/10000040/tasks' },
+            { body: tasksAndSors }
+          )
+        })
+        .as('taskListRequest')
+
+      cy.intercept(
+        {
+          method: 'GET',
+          path:
+            '/api/schedule-of-rates/codes?tradeCode=DE&propertyReference=00012345&contractorReference=SCC&showAdditionalTrades=true',
+        },
+        { fixture: 'scheduleOfRates/codes.json' }
+      ).as('sorCodesRequest')
+    })
+
+    it('allows adding multiple SORs', () => {
+      cy.visit('/work-orders/10000040/update')
+
+      cy.wait(['@taskListRequest', '@workOrder'])
+
+      cy.get('#repair-request-form').within(() => {
+        cy.contains('+ Add multiple SOR codes').click()
+      })
+
+      cy.get('#repair-request-form').should('not.exist')
+
+      cy.contains('Add multiple SOR codes')
+      cy.get('#adding-multiple-sors-form').within(() => {
+        cy.contains('Enter SOR codes as a list:')
+      })
```

Comment:
I didn't think of it before but please add an assertion within the same test for presence validation of SOR codes which submits the form and expects the validation error message.